libobs/obs-nix-wayland.c

```diff
@@ -19,7 +19,224 @@
 #include "obs-nix-platform.h"
 #include "obs-nix-wayland.h"
 
+#include <unistd.h>   // close
+#include <sys/mman.h> // mmap
+#include <string.h>   // strcmp
+
 #include <wayland-client.h>
+#include <xkbcommon/xkbcommon.h>
+
+// X11 only supports 256 scancodes, most keyboards dont have 256 keys so this should be reasonable.
+#define MAX_KEYCODES 256
+// X11 keymaps only have 4 shift levels, im not sure xkbcommon supports a way to shift the state into a higher level anyway.
+#define MAX_SHIFT_LEVELS 4
```

Comment:
Can this number be derived from the current keymap?

Reply:
These could be derived but if we consider non-X11 compatible both keycodes and shift levels have no limit. So such a simple implementation may not be feasible. Only supporting X11 compatible keymaps seems like a nice middle ground of simplicity and supporting all existing (x11 compatible) layouts.